runtime/jcl/common/jclexception.c

```diff
@@ -146,24 +144,26 @@ getStackTraceIterator(J9VMThread * vmThread, void * voidUserData, UDATA bytecode
 			}
 
 			/* Fill in module name and version */
-			if (j2seVersion >= J2SE_V11) {
-				J9Module *module = NULL;
-				if (classLoader != NULL) {
+			if (JAVA_SPEC_VERSION >= 11) {
+				if (NULL != classLoader) {
 					j9object_t classLoaderName = J9VMJAVALANGCLASSLOADER_CLASSLOADERNAME(vmThread, classLoader->classLoaderObject);
 					J9VMJAVALANGSTACKTRACEELEMENT_SET_CLASSLOADERNAME(vmThread, element, classLoaderName);
 				}
 				if (J9_ARE_NO_BITS_SET(vm->runtimeFlags, J9_RUNTIME_JAVA_BASE_MODULE_CREATED)) {
-					string = mmfns->j9gc_createJavaLangString(vmThread, (U_8 *)JAVA_BASE_MODULE, LITERAL_STRLEN(JAVA_BASE_MODULE), J9_STR_XLAT);
-					if (string == NULL) {
+					const char* javaModuleVersion = J9_STR(JAVA_SPEC_VERSION);
+					UDATA verLength = LITERAL_STRLEN(javaModuleVersion);
```

Comment:
LITERAL_STRLEN only works on literals, not variables. It uses sizeof().

Reply:
siezeof is also wrong if used on the variable. LITERAL_STRLEN on JAVA_SPEC_VERSION is correct.

Reply:
A better alternative would be to define a macro for the stringized version, and use that instead of making local variables at all. Then the literal code will work, and is the most performant.

Reply:
Added a macro `J9_STR_LITERAL_LEN(x)` as `LITERAL_STRLEN(J9_STR(x))`.

Reply:
Not a fan of that approach - extremely confusing name. What I suggest is defining `JAVA_SPEC_VERSION_STRING` as `J9_STR(JAVA_SPEC_VERSION)`. I would put this in the same place as `JAVA_SPEC_VERSION` is defined rather than putting it here.

Reply:
It turns out that `JAVA_SPEC_VERSION_STRING` already is defined at https://github.com/eclipse-openj9/openj9/blob/09971ec709ccfd48bea8b129fcfb8d0dc03f20c5/runtime/include/j9cfg.h.in#L45-L46
Updated PR accordingly.